src/app/shared/components/top-bar/top-bar.component.ts

```diff
@@ -10,14 +11,19 @@ import { ListService } from '../list/list.service';
 })
 export class TopBarComponent {
   constructor(
-    @Optional() private listService: ListService,
+    private sidebarContainerService: SidebarContainerService,
     private activatedRoute: ActivatedRoute,
+    @Optional() private listService: ListService,
   ) {}
 
   public get sidebarOpen(): boolean {
     return this.listService ? this.listService.hasSelected() && this.showSidebarForSG() : false;
   }
 
+  public get right() {
+    return this.sidebarOpen ? this.sidebarContainerService.width.getValue() : 0;
```

Comment:
use `width$ = this.sidebarContainerService.width.asObservable().pipe(map(width => this.sidebarOpen ? width : 0))` instead of getter